### PR TITLE
Add a README.md for benchmarking infra

### PR DESCRIPTION
**Pull request by guangy10**

### CODE REVIEW

**Review by mergennachin — APPROVED** [review]

File: extension/benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+> **Disclaimer:** The infrastructure is new and experimental. We're working on improving its accessibility and stability over time. The models performance tuning is still a work in progress. The benchmark numbers presented in this document are not vetted yet and should be treated as preliminary results.
+
+
+## Preliminary Benchmark Results
+
+Below is a table summarizing some example data points obtained via the infra. These numbers represent model load time and average inference latency across different platforms and backends.
+
+| Model                 | Backend      | Model Load Time (ms) | Avg Inference Latency (ms) | Device                 |
+|-----------------------|--------------|----------------------|----------------------------|------------------------|
+| DeepLabV3 (dl3)       | XNN Q8       | [90.616](https://github.com/pytorch/executorch/actions/runs/10875550238/job/30218985758)               | [666.219](https://github.com/pytorch/executorch/actions/runs/10875550238/job/30218985758)                    | Samsung S22            |
```
Comment (huydhn) [inline extension/benchmark/README.md:26]: Just FYI, the GitHub link here expires after 90 days, so let's work on getting the dashboard ready before that date :)

File: extension/benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## High-Level Diagram
+
+![Benchmarking Infrastructure](../../docs/source/_static/img/benchmark-infra.png)
+
+
+## Scheduling On-Demand Benchmarking
+
+Users can schedule a benchmarking workflow on a pull request through GitHub Actions using the workflow dispatch UI. Follow the steps below to trigger benchmarking:
```
Comment (huydhn) [inline extension/benchmark/README.md:58]: Not sure if we want to be clear here, only people with write permission to the repo can run the workflow.

File: extension/benchmark/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+1. Access `pytorch/executorch` repository on GitHub and navigate to the "Actions" tab.
+2. Select `android-perf` or `apple-perf` workflow from the list of workflows.
+3. Click "Run workflow" and fill in the required parameters for the model you want to benchmark, e.g. branch name, model name and delegate, and device pool, etc.
+
+
+## Retrieving Benchmark Results
+
+Currently, retrieving benchmark results involves manually extracting the `benchmark_results.txt` from the `Customer_Artifacts.zip` stored on AWS S3 from the benchmarking job. This process is not yet streamlined. We are working on simplifying this process and linking the results directly to the dashboard, which will be available soon.
```
Comment (huydhn) [inline extension/benchmark/README.md:66]: It should be `benchmark_results.json` now, I have get rid of the txt file

**Review by huydhn — APPROVED** [review]

File: extension/benchmark/README.md (hunk 2 above)
Comment (guangy10) [reply extension/benchmark/README.md:58]: Will add it to the readme. Having write permission will limit the usage of it, especially for backend vendors. Except for explicitly grant write permissions to individuals, is there other more convenient way for vendors to start dogfooding?

Comment (huydhn) [reply extension/benchmark/README.md:58]: There are similar questions/requests to GH from what I see https://github.com/orgs/community/discussions/49688.  We could consider building something like `@pytorchbot` here, for example, posting a comment like `@executorchbot benchmark` to run perf benchmark.  There are pros and cons, but that's a familiar approach that PyTorch is using

**Pull request closed without merging**